wolfcrypt/src/memory.c

```diff
@@ -514,20 +514,39 @@ void* wolfSSL_Realloc(void *ptr, size_t size)
 struct wc_Memory {
     byte*  buffer;
     struct wc_Memory* next;
+#ifdef WOLFSSL_LEAN_STATIC_MEMORY
+    /* lean static memory is assumed to be under 65k */
+    word16 sz;
+#else
     word32 sz;
+#endif
+#ifdef WOLFSSL_DEBUG_MEMORY
+    word16 szUsed;
+#endif
 };
 
 
+#ifdef WOLFSSL_DEBUG_MEMORY_CALLBACK
```

Comment:
Can `WOLFSSL_DEBUG_MEMORY_CALLBACK` be used on its own without static memory?

Reply:
It would be nice if it could:

```
./configure CFLAGS="-DWOLFSSL_DEBUG_MEMORY_CALLBACK" && make
...
examples/client/client.c:1871:14: error: use of undeclared identifier 'WOLFSSL_DEBUG_MEMORY_ALLOC'
        case WOLFSSL_DEBUG_MEMORY_ALLOC:
             ^
examples/client/client.c:1884:14: error: use of undeclared identifier 'WOLFSSL_DEBUG_MEMORY_FAIL'
        case WOLFSSL_DEBUG_MEMORY_FAIL:
             ^
examples/client/client.c:1888:14: error: use of undeclared identifier 'WOLFSSL_DEBUG_MEMORY_FREE'; did you mean 'WOLFSSL_BIO_MEMORY'?
        case WOLFSSL_DEBUG_MEMORY_FREE:
             ^~~~~~~~~~~~~~~~~~~~~~~~~
             WOLFSSL_BIO_MEMORY
./wolfssl/ssl.h:473:5: note: 'WOLFSSL_BIO_MEMORY' declared here
    WOLFSSL_BIO_MEMORY = 4,
    ^
examples/client/client.c:1892:14: error: use of undeclared identifier 'WOLFSSL_DEBUG_MEMORY_INIT'
        case WOLFSSL_DEBUG_MEMORY_INIT:
             ^
4 errors generated.
```

Reply:
Maybe combine it with WOLFSSL_DEBUG_MEMORY? Instead of a new macro.

Reply:
I take that back. The build am using it on does not like `_func_` so it would not build with `WOLFSSL_DEBUG_MEMORY` and `WOLFSSL_DEBUG_MEMORY_CALLBACK`.

Reply:
I left it as --enable-staticmemory only for the callback. The reasoning behind that was if using non static memory than users can override XMALLOC/XFREE to achieve close to the same thing at about the same effort as creating a callback function.